### PR TITLE
Clean up apt cache files in Dockerfile

### PR DESCRIPTION
**Pull request by PeterDaveHello**
This will help minimize the size of released Docker images ;)

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -77,6 +77,7 @@ RUN apt-get update && \
         tzdata \
         libreadline8 \
         tini && \
+    rm -rf /var/lib/apt/lists/* && \
```
Comment (nschonni) [inline Dockerfile:80]: There is a comment and haldolint ignore up on 58 that probably needs to be resolved

**Comment by vmstan** [comment]
What does the reduction look like?

File: Dockerfile (hunk 1 above)
Comment (PeterDaveHello) [reply Dockerfile:80]: oops, didn't notice DL3009, it's interesting to ignore that though

**Comment by PeterDaveHello** [comment]
@renchap will this be something acceptable once the conflict was resolved?

**Comment by renchap** [comment]
We intend to merge a large Dockerfile rework soon, most probably #26850

It would probably be best to ensure this change is included in it.

**Comment by PeterDaveHello** [comment]
Wow, that's a very huge PR, maybe some smaller PRs can land first, to make the review process easier :sweat_smile: ?

**Comment by vmstan** [comment]
The apt data is stored in Docker caches now and so clearing them would not be needed.

**Comment by vmstan** [comment]
These changes have been incorporated with the merger of #26850 

Closing this PR, welcoming future improvements to the new version.

**Pull request closed without merging**